docs/design/security/unix-tmp.md

```diff
@@ -0,0 +1,45 @@
+
+# Unix temporary files
+
+The Unix support for temporary files is different from the Windows model and developers who
+are used to Windows may inadvertantly create security risk if they use the same practices on Unix.
+
+Most notably, the Windows model for temporary files is that the operating system provides each user with a *unique*, *user-owned* temporary directory.
+Moreover, all Windows users, including the service and system users, have designated user folders, including temporary folders.
+
+The Unix model is very different. The temp directory, assuming there is one, is often a global folder (except on MacOS).
+If possible, prefer a library function like `GetTempPath()` to find the folder. Otherwise,
+the `TMPDIR` environment variable is used to store the location of this folder. This variable is
+widely used and supported, but it is not mandatory for all Unix implementations. It should be the preferred
+mechanism for finding the Unix temporary folder if a library method is not available. It will commonly
+point to either the `/tmp` or `/var/tmp` folder. These folders are not used for MacOS, so it is not recommended
+to use them directly.
+
+Because the temporary directory is often global, any use of the temp directory should be carefully
+considered. In general, the best use of the temp directory is for programs which,
+
+1. Will create the temporary file during their process execution
+1. Do not depend on predictable temporary file/folder names
+1. Will not access the file after the process exits
+
+In these cases, the process can create a file or files with
+  1. A pseudorandom name, unlikely to cause collisions
+  1. Permissions which restrict all access to owner-only, i.e. 700
+
+Any other use needs to be carefully audited, particularly if the temporary file is intended for use across
+multiple processes. Some considerations:
+
+- **Never** write files with global access permissions
+- **Always** verify that the owner of the file is the current user and that the permissions
```

Comment:
.NET doesn't have an API to verify file ownership.

Reply:
In that case I think P/Invoking into `stat` is necessary.

Reply:
cc @stephentoub on this one. Verifying file ownership is a super critical part of the unix security model. It's the reason why only root can change the ownership of files (you can't give ownership of your files to another user). It might be worth some sort of .NET API.

Reply:
In most cases you don't need to verify ownership. If you can open the dir/file you know you are permitted, and you can check the permissions to verify they are not too permissive.